File: hw/1422/1422.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(int num) {
    if (num <= 1) {
        return false;
    }
    
    for(int i = 2; i <= sqrt(num); i++) {
        if (num % i == 0) {
            return false;
        }
    }
    
    return true;
}

int main() {
    setlocale(LC_ALL, "");
    int n;
    cout << "Введите количество чисел: ";
    cin >> n;
    
    int count = 0;
    for (int i = 0; i < n; i++) {
        int num;
        cout << "Введите число " << i+1 << ": ";
        cin >> num;
        
        if (num > 0 and log(num) / log(5) == trunc(log(num) / log(5))) {
            count++;
        }
    }
    
    cout << "Количество чисел, являющихся степенями пятерки: " << count << endl;
    return 0;
}

File: hw/1422/1422.py
import math

# Функция для проверки числа на простоту
def is_prime(num):
    if num <= 1:
        return False
    
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    
    return True

n = int(input("Введите количество чисел: "))
count = 0

for i in range(n):
    num = int(input(f"Введите число {i+1}: "))
    
    # Проверяем, является ли число степенью пятерки
    if num > 0 and math.log(num, 5).is_integer():
        count += 1

print(f"Количество чисел, являющихся степенями пятерки: {count}")
